akka-docs/rst/scala/code/docs/persistence/query/MyEventsByTagPublisher.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package docs.persistence.query
+
+import akka.actor.Props
+import akka.persistence.PersistentRepr
+import akka.persistence.query.EventEnvelope
+import akka.serialization.SerializationExtension
+import akka.stream.actor.ActorPublisher
+import akka.stream.actor.ActorPublisherMessage.{ Cancel, Request }
+
+import scala.annotation.tailrec
+import scala.concurrent.duration.FiniteDuration
+
+object MyEventsByTagPublisher {
+  def props(tag: String, offset: Long, refreshInterval: FiniteDuration): Props =
+    Props(new MyEventsByTagPublisher(tag, offset, refreshInterval))
+
+  private case object Continue
+}
+
+//#events-by-tag-publisher
+class MyEventsByTagPublisher(tag: String, offset: Long, refreshInterval: FiniteDuration)
+  extends ActorPublisher[EventEnvelope] {
+  import MyEventsByTagPublisher._
+
+  private val limit = 1000
+
+  private var currentId = 0L
+  var buf = Vector.empty[EventEnvelope]
+
+  import context.dispatcher
+  val continueTask = context.system.scheduler.schedule(
+    refreshInterval, refreshInterval, self, Continue)
+
+  override def postStop(): Unit = {
+    continueTask.cancel()
+  }
+
+  def receive = {
+    case _: Request | Continue ⇒
+      query()
+      deliverBuf()
+
+    case Cancel ⇒
+      context.stop(self)
+  }
+
+  def query(): Unit =
+    if (buf.isEmpty) {
+      try {
+        // Could be an SQL query, for example:
+        //      "SELECT id, persistent_repr FROM journal WHERE tag = like ? and " +
```

Comment:
`= like` is not correct; could we show some actual code instead of `???` without pulling in all the internets as dependencies?